src/api/python.c

```diff
@@ -1,48 +1,63 @@
 #include "core/core.h"
 
-#include "pocketpy_c.h"
 #include <stdio.h>
 #include <string.h>
+#include "pocketpy.h"
+#include "pocketpy/objects/base.h"
 
+//!!! search NOTICE for v1 remains !!!
+
+/***************DEBUG SESSION*************/
+//cmake -A x64 -DCMAKE_BUILD_TYPE=MinSizeRel -DBUILD_SDLGPU=On -DBUILD_STATIC=On -DBUILD_WITH_ALL=Off -DBUILD_WITH_PYTHON=On ..
+//for debug
+//cmake -G "Visual Studio 16 2019" -A x64 -DCMAKE_BUILD_TYPE=MinSizeRel -DBUILD_SDLGPU=On -DBUILD_WITH_ALL=On ..
+//for release
 extern bool parse_note(const char* noteStr, s32* note, s32* octave);
+static bool py_throw_error(tic_core* core, const char* msg);
 
-struct CachedNames{
-    pkpy_CName _tic_core;
-    pkpy_CName len;
-    pkpy_CName __getitem__;
-    pkpy_CName TIC;
-    pkpy_CName BOOT;
-    pkpy_CName SCN;
-    pkpy_CName BDR;
-    pkpy_CName MENU;
+struct names
+{
+    py_Name _tic_core;
+    py_Name len;
+    py_Name __getitem__;
+    py_Name TIC;
+    py_Name BOOT;
+    py_Name SCN;
+    py_Name BDR;
+    py_Name MENU;
 } N;
 
-static void pkpy_setglobal_2(pkpy_vm* vm, const char* name){
-    pkpy_setglobal(vm, pkpy_name(name));
+//set value of "name" to stack.top
+static bool pkpy_setglobal_2(const char* name)
+{
+    py_setglobal(py_name(name), py_peek(-1));
 }
 
-static bool get_core(pkpy_vm* vm, tic_core** core)
+static bool get_core(tic_core** core)
```

Comment:
Change this function to `tic_core* get_core()` and it must return a valid pointer.